apps/snitch_core/lib/core/data/schema/order_address.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Snitch.Data.Schema.OrderAddress do
+  @moduledoc false
+
+  use Snitch.Data.Schema
+
+  alias Snitch.Data.Schema.{Country, State}
+  alias Snitch.Repo
+
+  @required_fields ~w(first_name last_name address_line_1 city zip_code country_id)a
+  @cast_fields ~w(phone alternate_phone state_id)a ++ @required_fields
+
+  embedded_schema do
+    field(:first_name, :string)
+    field(:last_name, :string)
+    field(:address_line_1, :string)
+    field(:address_line_2, :string)
+    field(:city, :string)
+    field(:zip_code, :string)
+    field(:phone, :string)
+    field(:alternate_phone, :string)
+
+    field(:state_id, :integer)
+    field(:country_id, :integer)
+  end
+
+  def changeset(%__MODULE__{} = address, params) do
+    address
+    |> cast(params, @cast_fields)
+    |> validate_required(@required_fields)
+    |> validate_length(:address_line_1, min: 10)
+    |> assoc_state_and_country()
+  end
+
+  defp struct_to_map(struct) do
+    struct
+    |> Map.from_struct()
+    |> Map.delete(:__meta__)
+  end
+
+  defp assoc_state_and_country(
```

Comment:
@oyeb we have similar code in multiple places for address ? Will have to see if it can be extracted to an address validator across the system.

Reply:
I think we can get rid of the existing table-backed Address schema altogether, so we'll have only this module in the end.

Reply:
`existing table-backed Address` ? 

We'll need the same functions in stock locations also. and user address too. So we can extract state country validation to a validator.

Reply:
No, we don't. We'll just embed `Snitch.Data.Schema.OrderAddress` everywhere (after renaming it to `EmbeddedAddress` perhaps).

Reply:
hmm